usage/src/main/java/org/killbill/billing/usage/api/BaseUserApi.java

```diff
@@ -73,18 +76,75 @@ public BaseUserApi(final OSGIServiceRegistration<UsagePluginApi> pluginRegistry)
             // First plugin registered, returns result -- could be empty List if no usage was recorded.
             if (result != null) {
 
+                final DebugMap debugMap = new DebugMap(startDate, endDate, logger);
                 for (final RawUsageRecord cur : result) {
                     if (cur.getDate().compareTo(startDate) < 0 || cur.getDate().compareTo(endDate) >= 0) {
                         logger.warn("Usage plugin returned usage data with date {}, not in the specified range [{} -> {}[",
                                     cur.getDate(), startDate, endDate);
                     }
+                    debugMap.add(cur);
                 }
+                debugMap.logDebug();
                 return result;
             }
         }
         // All registered plugins returned null
         return null;
     }
 
+    private static class DebugMap {
+
+        private final Map<UUID, List<RawUsageRecord>> perSubscriptionRecords;
+        private final Logger logger;
```

Comment:
You might not have to pass the `logger` -- I would think it would be visible in here.

Reply:
I think because the class is `static` I need to -- but I have not checked.